### PR TITLE
feat(waf/group): add a resource to manage the WAF instance groups

### LINKED ISSUE #1626: [WAF] add a resource to manage the WAF instance group
Add a resource to manage the WAF instance group.
Resource name : `huaweicloud_waf_instance_group`.

### PR DESCRIPTION
**Pull request by chengxiangdong**
**What this PR does / why we need it**:

CBG needs to create ELB mode WAF instances, and an instance group must be created before it creation.
So, add a resource to manage the WAF instance groups.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1626

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccWafInstanceGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccWafInstanceGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccWafInstanceGroup_basic
=== PAUSE TestAccWafInstanceGroup_basic
=== CONT  TestAccWafInstanceGroup_basic
--- PASS: TestAccWafInstanceGroup_basic (58.62s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       58.719s

```

### CODE REVIEW

File: docs/resources/waf_instance_group.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  vpc_id = var.vpc_id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) The region in which to create the instance group.
+  If omitted, the provider-level region will be used. Changing this setting will create a new certificate.
```
Comment (niuzhenguo) [inline docs/resources/waf_instance_group.md]: s/certificate/resource

File: huaweicloud/services/waf/resource_huaweicloud_waf_instance_group.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 256),
+			},
+			"body_limit": {
```
Comment (niuzhenguo) [inline huaweicloud/services/waf/resource_huaweicloud_waf_instance_group.go:54]: do we need to add an update action for such arguments in Create function after the group created?

Comment (ShiChangkuo) [reply huaweicloud/services/waf/resource_huaweicloud_waf_instance_group.go:54]: the limitation can not work in a pipeline. can we add a new resource or ask the API side to fix it?

File: docs/resources/waf_instance_group.md (hunk 1 above)
Comment (chengxiangdong) [reply docs/resources/waf_instance_group.md]: done

File: huaweicloud/services/waf/resource_huaweicloud_waf_instance_group.go (hunk 2 above)
Comment (chengxiangdong) [reply huaweicloud/services/waf/resource_huaweicloud_waf_instance_group.go:54]: negotiating with the API provider, set to read-only first

**Pull request merged**